#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kinematics/tinkerbots/protos/TinkerbotsBase.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kinematics/tinkerbots/protos/TinkerbotsPivot.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kinematics/tinkerbots/protos/TinkerbotsTwister.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kinematics/tinkerbots/protos/TinkerbotsMotor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kinematics/tinkerbots/protos/TinkerbotsCube.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kinematics/tinkerbots/protos/TinkerbotsAxle.proto"
EXTERNPROTO "../protos/TinkerbotsLightSensor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kinematics/tinkerbots/protos/TinkerbotsWheel.proto"
EXTERNPROTO "../protos/TinkerbotsDistanceSensor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/kinematics/tinkerbots/protos/TinkerbotsCubieBoxWithRoundSlots.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.2516018518858532 0.13251817722134784 0.9587155161118135 2.206398807373521
  position 1.4920947654564785 -1.7563008848775379 1.2895007968803465
  follow "Tinkerbots"
}
TexturedBackground {
}
TexturedBackgroundLight {
  luminosity 0
}
RectangleArena {
  floorSize 2 1
}
TinkerbotsBase {
  translation 0.65767 -0.309078 0.0999843
  rotation 0.5773472691841857 0.5773512691860505 -0.5773522691865168 2.09439
  controller "Obstacle_Robot"
  upSlot TinkerbotsPivot {
    rotation 0 0 1.0000000000000002 1.5707999999999998
    slot TinkerbotsMotor {
      frontSlot TinkerbotsAxle {
        endSlot TinkerbotsWheel {
        }
      }
      backSlot TinkerbotsAxle {
        name "axle(1)"
        endSlot TinkerbotsWheel {
        }
      }
    }
  }
  downSlot TinkerbotsTwister {
    slot TinkerbotsCube {
      leftSlot TinkerbotsLightSensor {
        rotation 0 0 1.0000000000000002 1.5707999999999998
        name "left light sensor"
        upSlot TinkerbotsDistanceSensor {
          name "left distance sensor"
          numberOfRays 2
          aperture 0.5
          backDSlot TinkerbotsCubieBoxWithRoundSlots {
            translation -6.4763690379141086e-12 -4.4078460467215906e-07 0
            rotation 0 0 1 6.283170614392848
            axisSlot TinkerbotsAxle {
              rotation 0 1 0 3.1415
              endSlot TinkerbotsWheel {
              }
            }
          }
        }
      }
      rightSlot TinkerbotsLightSensor {
        rotation 0 0 -1.0000000000000002 1.5707999999999998
        name "right light sensor"
        upSlot TinkerbotsDistanceSensor {
          name "right distance sensor"
          numberOfRays 2
          aperture 0.5
          backDSlot TinkerbotsCubieBoxWithRoundSlots {
            translation -6.476369035984995e-12 4.407846045408627e-07 0
            rotation 0 0 -1 6.283170614392848
            axisSlot TinkerbotsAxle {
              rotation 0 1 0 3.1415
              endSlot TinkerbotsWheel {
              }
            }
          }
        }
      }
    }
  }
  extensionSlot [
    GPS {
    }
    Accelerometer {
    }
  ]
}
